lib/tests/mocha/unit/concurrency.spec.ts

```diff
@@ -29,29 +33,46 @@ describe('concurrency', () => {
     describe(`anyPool(${n})`, () => {
       it('should resolve with the first resolved promise', async () => {
         const startTime = Date.now();
+        const started = new Set<string>();
         const promises = {
-          a: new Promise((resolve) => setTimeout(() => resolve(1), 500)),
-          b: new Promise((resolve) => setTimeout(() => resolve(2), 50)),
-          c: new Promise((resolve) => setTimeout(() => resolve(3), 1500)),
+          a: async () => {
+            started.add('a');
+            await sleep(500);
+            return 1;
+          },
+          b: async () => {
+            started.add('b');
+            await sleep(50);
+            return 2;
+          },
+          c: async () => {
+            started.add('c');
+            await sleep(1500);
```

Comment:
how did you pick these values? for the sleeps?

Reply:
They need to be different enough so they are probably almost never going to flake in CI. I probably am being too sensitive on the assertion, actually. I probably should somehow randomize or round-robin the order and do it several times, if I want to be safe